### PR TITLE
upspin: init at unstable-2023-02-05

### PR DESCRIPTION
**Pull request by orthros**
###### Description of changes
Adds new package upspin. 

Homepage: https://upspin.io

Description from their website:

Upspin is an [open-source project](https://github.com/upspin/upspin) that comprises two main design elements:

1. a set of protocols enabling secure, federated sharing using a global naming system; and
1. reference implementations of tools and services that demonstrate the capabilities.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ocfox** [comment]
You forgot to add this package to the list.

**Review by ocfox — COMMENTED** [review]
You forgot to add this package to the list.

**Comment by orthros** [comment]
Please forgive my ignorance: what do you mean by "the list"? It has been a while since I've added a new package :bow:

**Comment by ocfox** [comment]
The top-level entry, at [pkgs/top-level/all-packages.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix). You can refer to other PR.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -39658,4 +39660,5 @@ with pkgs;
   udict = callPackage ../applications/misc/udict { };
 
   duden = callPackage ../applications/misc/duden { };
+
```
Comment (ocfox) [inline pkgs/top-level/all-packages.nix:39663]:
```suggestion
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -39535,6 +39535,8 @@ with pkgs;
 
   go-license-detector = callPackage ../development/tools/misc/go-license-detector { };
 
+  upspin = callPackage ../tools/filesystems/upspin { };
+
```
Comment (ocfox) [inline pkgs/top-level/all-packages.nix:39539]: Place in alphabetical order in the approximate position.

File: pkgs/tools/filesystems/upspin/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    sha256 = "sha256-1pFDJSCUDKn4CTAg3wdB8oYPyrmd8B62zNl3m5YAqVM=";
+  };
+
+  vendorSha256 = "sha256-Jl++FvKyqz5WFa/Eoly+UnFsoC9Qwdaizhkq6LyJ+XQ=";
+
+  # No upstream tests
+  doCheck = false;
+
+  meta = with lib; {
+    description = "A set of protocols and reference implementations that can be used to join things like file systems and other storage services to a global name space.";
```
Comment (ocfox) [inline pkgs/tools/filesystems/upspin/default.nix:20]:
```suggestion
    description = "A set of protocols and reference implementations that can be used to join things like file systems and other storage services to a global name space";
```
This description may be a little long, it's better if you can streamline it.

File: pkgs/tools/filesystems/upspin/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{ buildGoModule, fetchFromGitHub, lib }:
+
+buildGoModule rec {
+  pname = "upspin";
+  version = "unstable-2023-02-05";
```
Comment (ocfox) [inline pkgs/tools/filesystems/upspin/default.nix:5]:
```suggestion
  version = "2023-02-05";
```

**Review by ocfox — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (orthros) [reply pkgs/top-level/all-packages.nix]: Done.

File: pkgs/tools/filesystems/upspin/default.nix (hunk 4 above)
Comment (ocfox) [reply pkgs/tools/filesystems/upspin/default.nix]: This is my mistake, forget it.
See https://nixos.org/manual/nixpkgs/unstable/#sec-package-naming

Comment (orthros) [reply pkgs/tools/filesystems/upspin/default.nix:5]: Forgotten.

File: pkgs/tools/filesystems/upspin/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  # No upstream tests
+  doCheck = false;
+
+  meta = with lib; {
+    description = "A global name space for storing data akin to a filesystem";
+    homepage = "https://upspin.io";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ orthros ];
+    platforms = platforms.linux;
+    mainProgram = "upspin";
```
Comment (doronbehar) [inline pkgs/tools/filesystems/upspin/default.nix:25]: Not needed if this is also `pname`.

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -11360,6 +11360,11 @@
     githubId = 357005;
     name = "Marco Orovecchia";
   };
+  orthros = {
```
Comment (doronbehar) [inline maintainers/maintainer-list.nix:11363]: Should be in a separate commit.

**Review by doronbehar — COMMENTED** [review]
Looks almost perfect to me.

File: maintainers/maintainer-list.nix (hunk 6 above)
Comment (orthros) [reply maintainers/maintainer-list.nix:11363]: Sounds good!

Just for my clarity: a separate commit in this PR, or a separate commit in a second PR?

**Comment by doronbehar** [comment]
In this PR, with the message:

maintainers: add orthros

**Review by doronbehar — APPROVED** [review]
